eventsourcing/src/main/java/org/axonframework/eventsourcing/annotation/EventCriteriaBuilder.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2010-2025. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.eventsourcing.annotation;
+
+import org.axonframework.eventsourcing.eventstore.EventCriteria;
+import org.axonframework.modelling.command.annotation.TargetEntityId;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Annotation to indicate that a method can be used to resolve the {@link TargetEntityId} to {@link EventCriteria}
```

Comment:
```suggestion
 * Annotation to indicate that a method can be used to resolve the {@link EventCriteria} based on {@link TargetEntityId} 
```
It's easier to understand for me, but up to you.